### PR TITLE
New LCD emulation for devof12e

### PR DESCRIPTION
**Pull request by howard0su**
Use a low level emulatioin for LCD controller in devof12e so that we can
test LCD.c code in emulator.

While I am here, cleanup LCD.c to only contains the high level logic and leave the logic interfacing TW8816 chip to tw8816.c

### CODE REVIEW

**Comment by howard0su** [comment]
This is ongoing work to support localization for devof12e.

**Comment by howard0su** [comment]
tested on devof12e and emu_devof12e.

**Comment by TheRealMoeder** [comment]
This seems to have broken nightlies...

File: src/target/devof12e/tw8816.c [hunk 1]
```diff
@@ -29,19 +31,230 @@
...
+}
+
+static u32 LCD_ReadReg(unsigned reg)
+{
+    u8 val;
+    for (int i = 0; i < 2; i++) {
+        if (! I2C1_ReadBuffer(0x45, &val, reg, 1))
+            break;
+    }
+    return val;
```
Comment (TheRealMoeder) [inline src/target/devof12e/tw8816.c]: We still get a compiler warning here about `val` possibly being uninitialized

Comment (howard0su) [reply src/target/devof12e/tw8816.c]: i will fix it in #503

Comment (howard0su) [reply src/target/devof12e/tw8816.c]: already fixed in master.

**Pull request merged**